package/MDAnalysis/topology/LAMMPSParser.py

```diff
@@ -584,56 +771,133 @@ def _parse_box(self, header):
 
 
 class LammpsDumpParser(TopologyReaderBase):
-    """Parses Lammps ascii dump files in 'atom' format
+    """Parse a LAMMPS ascii dump files in any format.
+
+    `LammpsDumpParser` can handle any standard or custom LAMMPS dump files
+    provided that the columns describtors (keywords similar to the attributes
+    in the atom_style argument in DATAParser) are included in the “ITEM: ATOMS”
+    line as described in dump_ documentation.
 
-    Only reads atom ids.  Sets all masses to 1.0.
+    However, `LammpsDumpParser` only parses `id`, `type`, `resid` ('mol' in
+    dump_ doc), `charge` ('q' in dump_ doc), `mass` attributes. The `resid`,
+    `charge` and `mass` attributes are optional and any other attribute will be
+    ignored.
+
+    Sets all masses to 1.0 if there is no `mass` info in the parsed dump file.
 
     .. versionadded:: 0.19.0
+
+    .. _dump: http://lammps.sandia.gov/doc/dump.html
     """
     format = 'LAMMPSDUMP'
 
     def parse(self, **kwargs):
+        # mol: resid
+        valid_attrs = ['id', 'mol', 'type', 'mass', 'q']
+        reqd_attrs = ['id', 'type']
+
         with openany(self.filename) as fin:
             fin.readline()  # ITEM TIMESTEP
             fin.readline()  # 0
 
             fin.readline()  # ITEM NUMBER OF ATOMS
-            natoms = int(fin.readline())
+            n_atoms = int(fin.readline())
 
             fin.readline()  # ITEM BOX
             fin.readline()  # x
             fin.readline()  # y
             fin.readline()  # z
 
-            indices = np.zeros(natoms, dtype=int)
-            types = np.zeros(natoms, dtype=object)
-            
-            fin.readline()  # ITEM ATOMS
-            for i in range(natoms):
-                idx, atype, _, _, _ = fin.readline().split()
-
-                indices[i] = idx
-                types[i] = atype
+            atom_ids = np.zeros(n_atoms, dtype=int)
+            types = np.zeros(n_atoms, dtype=object)
+
+            line = fin.readline()  # ITEM ATOMS
+            # column describtors comes after '# ITEM ATOMS ' as explained in
+            # http://lammps.sandia.gov/doc/dump.html
+            cols = line.split("# ITEM ATOMS ")[1].split()
```

Comment:
Hello! I was playing around with this code because I would like to use this functionality. I think the actual string is "ITEM: ATOMS ", not "# ITEM ATOMS", per dump files I have seen and the documentation page referred to here (there's this line in the docs: "The “ITEM: ATOMS” line in each snapshot lists column descriptors for the per-atom lines that follow."). Hopefully that helps!